service/matching/taskReader.go

```diff
@@ -171,6 +171,8 @@ Loop:
 				if !isReadBatchDone {
 					tr.Signal()
 				}
+				// sleep for a while when no tasks is available, otherwise too many queries
+				time.Sleep(time.Second)
```

Comment:
shouldn't need this sleep. If there is no task, the notifyC channel should have no more messages. If something keeps notifying the channel, then we have other problem that should be fixed.